### PR TITLE
refactor the playbook

### PR DESCRIPTION
**Pull request by arbabnazar**
@jfrappier please review. Thanks

### CODE REVIEW

File: emccode-rackhd-playbook-ubuntu.yml [hunk 1]
```diff
@@ -1,57 +1,67 @@
...
-    apt: upgrade=dist
-    command: apt-get -y autoremove
+    
+  - name: remove non-essential packages
+    command: >
+      apt-get -y autoremove
 
-  - name: Install required packages
-    apt: name=rabbitmq-server state=present
-    apt: name=mongodb state=present
```
Comment (arbabnazar) [inline emccode-rackhd-playbook-ubuntu.yml]: will not work like this, it has be to like this:
``yaml
   - name: Install required packages
     apt: name=rabbitmq-server state=present
   - apt: name=mongodb state=present
```

File: emccode-rackhd-playbook-ubuntu.yml [hunk 2]
```diff
@@ -1,57 +1,67 @@
...
-
-  - name: Install supplemental packages
-    apt: name=ansible state=present
-    apt: name=apt-mirror state=present
-    apt: name=amtterm state=present
+  - name: Install required and supplemental packages
+    apt: 
+      name: "{{ item }}"
+      state: present
+    with_items:
```
Comment (arbabnazar) [inline emccode-rackhd-playbook-ubuntu.yml]: that's much cleaner and recommended way.

File: emccode-rackhd-playbook-ubuntu.yml [hunk 3]
```diff
@@ -1,57 +1,67 @@
...
     command: sudo ln -s /var/mirrors/esxi <on-http directory>/static/http/vmware
-    command: sudo ln -s /var/mirrors/esxi <on-tftp directory>/static/tftp/vmware
-    command: sudo ln -s /var/mirrors/ubuntu_trusty <on-http directory>/static/http/ubuntu_trusty
-    command: sudo ln -s /var/mirrors/ubuntu/14.04/mirror/mirror.pnl.gov/ubuntu/ <on-http directory>/static/http/trusty
-    command: sudo ln -s /var/mirrors/centos <on-http directory>/static/http/centos
-    command: sudo ln -s /var/mirrors/suse <on-http directory>/static/http/suse
-
-  - name: Install supplementary files
-   apt: name=on-static-common state=present
+  - command: sudo ln -s /var/mirrors/esxi <on-tftp directory>/static/tftp/vmware
```
Comment (arbabnazar) [inline emccode-rackhd-playbook-ubuntu.yml]: this is the example that I have mentioned above on the `apt` task, but can also use the `with_items` here.

File: emccode-rackhd-playbook-ubuntu.yml [hunk 4]
```diff
@@ -1,57 +1,67 @@
...
+      - nodejs-legacy
+      - npm
+      - ansible
+      - apt-mirror
+      - amtterm
+      - on-static-common
 
   - name: Add official sensu repo
     copy:
-      src=files/monorail.list
```
Comment (arbabnazar) [inline emccode-rackhd-playbook-ubuntu.yml]: it exits inside the playbook directory not inside the `files` directory.

File: emccode-rackhd-playbook-ubuntu.yml [hunk 5]
```diff
@@ -1,57 +1,67 @@
 ---
 - hosts: **Update Hosts**
   remote_user: **Update Remote User**
-  sudo: yes
+  become: yes
```
Comment (arbabnazar) [inline emccode-rackhd-playbook-ubuntu.yml]: `sudo: yes` going to be deprecate soon and replace with `become: yes`

File: emccode-rackhd-playbook-ubuntu.yml (hunk 4 above)
Comment (jfrappier) [inline emccode-rackhd-playbook-ubuntu.yml]: Thanks, originally I had it in a files directory but missed that when I added. GitHub is currently being blocked by the corporate firewall so needed to manually add to GitHub

**Comment by jfrappier** [comment]
LGTM - thanks for the tips on cleaning up.

**Pull request merged**